sdks/java/io/solr/src/test/java/org/apache/beam/sdk/io/solr/SolrIOTestUtils.java

```diff
@@ -0,0 +1,138 @@
+  public static final long AVERAGE_DOC_SIZE = 25L;
+  public static final long MAX_DOC_SIZE = 35L;
+
+  static void createCollection(String collection, int numShards, int replicationFactor,
+      AuthorizedCloudSolrClient client)
+      throws Exception {
+    CollectionAdminRequest.Create create = new CollectionAdminRequest.Create()
+        .setCollectionName(collection)
+        .setNumShards(numShards)
+        .setReplicationFactor(replicationFactor)
+        .setMaxShardsPerNode(2);
+    client.process(create);
+  }
+
+  /** Inserts the given number of test documents into Solr. */
+  static void insertTestDocuments(String collection, long numDocs, AuthorizedCloudSolrClient client)
+      throws IOException {
+    List<SolrInputDocument> data = createDocuments(numDocs);
+    try {
+      UpdateRequest updateRequest = new UpdateRequest();
+      updateRequest.setAction(UpdateRequest.ACTION.COMMIT, true, true);
+      updateRequest.add(data);
+      client.process(collection, updateRequest);
+    } catch (SolrServerException e) {
+      throw new IOException("Failed to insert test documents in collection " + collection, e);
+    }
+  }
+
+
+  /** Delete given collection. */
+  static void deleteCollection(String collection, AuthorizedCloudSolrClient client)
+      throws IOException {
+    try {
+      CollectionAdminRequest.Delete delete = new CollectionAdminRequest.Delete()
+          .setCollectionName(collection);
+      client.process(delete);
+    } catch (SolrServerException e) {
+      throw new IOException(e);
+    }
+
+  }
+
+  /** Clear given collection. */
+  static void clearCollection(String collection, AuthorizedCloudSolrClient client)
+      throws IOException {
+    try {
+      UpdateRequest updateRequest = new UpdateRequest();
+      updateRequest.setAction(UpdateRequest.ACTION.COMMIT, true, true);
+      updateRequest.deleteByQuery("*:*");
+      client.process(collection, updateRequest);
+    } catch (SolrServerException e) {
+      throw new IOException(e);
+    }
+  }
+
+  /**
+   * Forces a commit of the given collection to
+   * make recently inserted documents available for search.
+   *
+   * @return The number of docs in the index
+   */
+  static long commitAndGetCurrentNumDocs(String collection, AuthorizedCloudSolrClient client)
+      throws IOException {
+    SolrQuery solrQuery = new SolrQuery("*:*");
+    solrQuery.setRows(0);
+    try {
+      UpdateRequest update = new UpdateRequest();
+      update.setAction(UpdateRequest.ACTION.COMMIT, true, true);
+      client.process(collection, update);
+
+      return client.query(new SolrQuery("*:*")).getResults().getNumFound();
+    } catch (SolrServerException e) {
+      throw new IOException(e);
+    }
+  }
+
+  /**
+   * Generates a list of test documents for insertion.
+   *
+   * @return the list of json String representing the documents
+   */
+  static List<SolrInputDocument> createDocuments(long numDocs) {
+    String[] scientists = {
+        "Einstein",
```

Comment:
Consider using the more diverse set of scientists from CassandraIOTest :)

      String[] scientists = {
          "Lovelace",
          "Franklin",
          "Meitner",
          "Hopper",
          "Curie",
          "Faraday",
          "Newton",
          "Bohr",
          "Galilei",
          "Maxwell"
      };